### ISSUE TITLE
Xija build fails on CentOS

### ISSUE THREAD

**Issue by javierggt** [sot/skare3#668]
**Description**

After tagging/commit, a build is expected. This one failed even though it says the linux build succeded: https://github.com/sot/xija/runs/2964566445

The relevant error is:
```
  File "/github/workspace/tmp/tmp4ccssd72/skare3/builds/xija_1625157227281/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/distutils/sysconfig.py", line 446, in _init_posix
    _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
ModuleNotFoundError: No module named '_sysconfigdata_x86_64_conda_linux_gnu'
```

**To Reproduce**

I reproduced the steps the builder follows:
```
docker login docker.pkg.github.com
docker pull docker.pkg.github.com/sot/skare3/centos7-builder:latest
docker run --rm  --name builder -v $GITHUB_WORKSPACE:/github/workspace -w /github/workspace \
    -e CONDA_PASSWORD -e GIT_USERNAME -e GIT_PASSWORD \
    docker.pkg.github.com/sot/skare3/centos7-builder:latest \
    sot/xija --force --tag 4.23.0
```

where the GITHUB_WORKSPACE, CONDA_PASSWORD, GIT_USERNAME and GIT_PASSWORD need to already be defined.

**Some context**

https://github.com/conda-forge/python-feedstock/issues/220

**Comment by javierggt** [comment]
I still do not know how to fix this, but this is my understanding thus far.

I can reproduce the import error by getting into the docker image, activating the ska-builder environment and doing `sysconfig.get_path("purelib")` in python.
```
>>> import sysconfig
>>> sysconfig.get_path("purelib")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ska/miniconda3/envs/ska-builder/lib/python3.8/sysconfig.py", line 521, in get_path
    return get_paths(scheme, vars, expand)[name]
  File "/home/ska/miniconda3/envs/ska-builder/lib/python3.8/sysconfig.py", line 511, in get_paths
    return _expand_vars(scheme, vars)
  File "/home/ska/miniconda3/envs/ska-builder/lib/python3.8/sysconfig.py", line 172, in _expand_vars
    _extend_dict(vars, get_config_vars())
  File "/home/ska/miniconda3/envs/ska-builder/lib/python3.8/sysconfig.py", line 559, in get_config_vars
    _init_posix(_CONFIG_VARS)
  File "/home/ska/miniconda3/envs/ska-builder/lib/python3.8/sysconfig.py", line 430, in _init_posix
    _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
ModuleNotFoundError: No module named '_sysconfigdata_x86_64_conda_linux_gnu'
```

This comes from an environment variable created by the `gcc_linux-64` package, I believe. At the other end, python is patched ([this is the patch](https://github.com/conda-forge/python-feedstock/blob/3.8/recipe/patches/0012-Add-support-for-_CONDA_PYTHON_SYSCONFIGDATA_NAME-if-.patch)) so this environment variable is understood and a correspondingly named module is imported.

Indeed, if I unset `_CONDA_PYTHON_SYSCONFIGDATA_NAME`, `sysconfig.get_path("purelib")` succeeds without error.

If I list all `sysconfg_*` modules, it is clear that the module is just not there:
```
(ska-builder)  ska /github/workspace/skare3>ls /home/ska/miniconda3/envs/ska-builder/lib/python3.8/_sysconfigdata*
/home/ska/miniconda3/envs/ska-builder/lib/python3.8/_sysconfigdata__linux_x86_64-linux-gnu.py
/home/ska/miniconda3/envs/ska-builder/lib/python3.8/_sysconfigdata__linux_x86_64-linux-gnu.py.orig
/home/ska/miniconda3/envs/ska-builder/lib/python3.8/_sysconfigdata_aarch64_conda_cos7_linux_gnu.py
/home/ska/miniconda3/envs/ska-builder/lib/python3.8/_sysconfigdata_i686_conda_cos6_linux_gnu.py
/home/ska/miniconda3/envs/ska-builder/lib/python3.8/_sysconfigdata_powerpc64le_conda_cos7_linux_gnu.py
/home/ska/miniconda3/envs/ska-builder/lib/python3.8/_sysconfigdata_x86_64_apple_darwin13_4_0.py
/home/ska/miniconda3/envs/ska-builder/lib/python3.8/_sysconfigdata_x86_64_conda_cos6_linux_gnu.py
```
but when `_CONDA_PYTHON_SYSCONFIGDATA_NAME` is not defined, then it loads `_sysconfigdata__linux_x86_64-linux-gnu`, which is given by:
```
            '_sysconfigdata_{abi}_{platform}_{multiarch}'.format(
            abi=sys.abiflags,
            platform=sys.platform,
            multiarch=getattr(sys.implementation, '_multiarch', '')
```

The problem seems to be that the environment variable set by the gcc package just does not correspond to what is expected by the sysconfig module.

**Comment by javierggt** [comment]
I checked that the expected file:
```
lib/python3.8/_sysconfigdata_x86_64_conda_linux_gnu.py
```
exists only in version 3.8.10, and we use 3.8.3. It is not there in the intermediate versions (3.8.5, 3.8.8). The conda package in https://anaconda.org/anaconda/python/files?version=3.8.3 is from a month ago.

Based on this, I am guessing that they modified all the gcc packages accordingly to use this file.

The solutions, other than an upstream fix, are:
- upgrade to python 3.8.10 in the build.
- change the build by adding a symbolic link or unsetting _CONDA_PYTHON_SYSCONFIGDATA_NAME, which is a hack and I still need to see how. I tried unsetting in the top build environment but the error still occurred.
- ~not use the docker image to build~

In addition, if we do not want to change packages in ska3-core, then we need to change the xija recipe to request the specific versions in the build environment (still need to check).

**Comment by taldcroft** [comment]
In theory I don't mind going to Python 3.8.10 in the build so long as it doesn't create a bunch of other changes. And is that just in the build or are you talking about updating the deployed Python as well? (I'm guessing the latter but not sure).

**Comment by javierggt** [comment]
I found an easy solution. One can just define the _CONDA_PYTHON_SYSCONFIGDATA_NAME variable and that takes precedence over the wrong name defined by default. I can just add it to the docker image, and that should fix the problems everywhere.

We would still have the issue that some package versions need to be specifically set, but at least we can build in whatever version of python we want.

**Comment by javierggt** [comment]
I will take this opportunity to make a new version of the builder, changing the conda channel from shiny to flight, and installing skare3_tools from there.

**Comment by javierggt** [comment]
scratch that... it does not work because conda creates the temporary build environment, and the environment variable is not propagated in there because gcc is installed in this temporary environment.

I tried installing gcc in the main build environment, and then for some reason conda build did not work. It just outputs `[y/N]` and does nothing regardless of the answer.

When I said to use python 3.8.10 I meant only for building. The ABI _should_ not change between patch versions (famous last words), so I figured that a packaged built with 3.8.10 would still run in 3.8.3 but who knows?
